components/proxy/src/test/kotlin/com/hotels/styx/Support.kt

```diff
@@ -149,3 +134,35 @@ fun failingMockObject() = mockk<RoutingObject> {
 fun mockObjectFactory(objects: List<RoutingObject>) = mockk<RoutingObjectFactory> {
     every { build(any(), any(), any()) } returnsMany objects
 }
+
+
+private val LOGGER = LoggerFactory.getLogger("ProxySupport")
+
+fun CompletableFuture<HttpResponse>.wait(debug: Boolean = false) = this.toMono()
+        .doOnNext {
+            if (debug) {
+                LOGGER.debug("${it.status()} - ${it.headers()} - ${it.bodyAs(UTF_8)}")
+            }
+        }
+        .block()
+
+fun CompletableFuture<LiveHttpResponse>.wait(debug: Boolean = false) = this.toMono()
+        .doOnNext {
+            if (debug) {
+                LOGGER.debug("${it.status()} - ${it.headers()}")
+            }
+        }
+        .block()
+
+fun Eventual<LiveHttpResponse>.wait(maxBytes: Int = 100*1024, debug: Boolean = false) = this.toMono()
+        .flatMap { it.aggregate(maxBytes).toMono() }
+        .doOnNext {
+            if (debug) {
+                LOGGER.info("${it.status()} - ${it.headers()} - ${it.bodyAs(UTF_8)}")
+            }
+        }
+        .block()
+
+val globalExecutor = Executors.newSingleThreadExecutor()
```

Comment:
Should we replace this with Runnable:run as discussed?

Reply:
👍